objects/src/notes/nullifier.rs

```diff
@@ -16,7 +18,7 @@ use crate::utils::{hex_to_bytes, HexParseError};
 /// - We cannot derive a note's hash from its nullifier, or a note's nullifier from its hash.
 /// - To compute the nullifier we must know all components of the note: serial_num, script_hash,
 ///   input_hash and asset_hash.
-#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
```

Comment:
I'm guessing we need this to put nullifiers as keys into a `BTreeMap`? In which circumstances are we doing this?

Reply:
Yes, it's for `BTreeMap`, as we want to use `Nullifier` instead of `RpoDigest`. We use it here: https://github.com/0xPolygonMiden/miden-node/blob/db51f392257abcf2fd5a703a15b830e4d8a5f7ab/block-producer/src/block_builder/prover/block_witness.rs#L35 and here: https://github.com/0xPolygonMiden/miden-node/blob/db51f392257abcf2fd5a703a15b830e4d8a5f7ab/proto/src/domain/transactions.rs#L26